napari/layers/points/points.py

```diff
@@ -928,6 +937,63 @@ def on_mouse_release(self, event):
                 self.selected_data = []
             self._set_highlight(force=True)
 
+    def _transform_color(
+        self, colors: ColorType, elem_name: str, default: str
+    ) -> np.ndarray:
+        """Helper method to return a ColorArray from an arbitrary user input."""
+        try:
+            transformed = transform_color(colors)
+        except (AttributeError, ValueError, KeyError):
+            warnings.warn(
+                f"The provided {elem_name} parameter contained illegal values, "
+                f"reseting all {elem_name} values to {default}."
+            )
+            transformed = transform_color(default)
+        else:
+            if (len(transformed) != 1) and (
+                len(transformed) != len(self.data)
+            ):
+                warnings.warn(
+                    f"The provided {elem_name} parameter has {len(colors)} entries, "
+                    f"while the data contains {len(self.data)} entries. Setting {elem_name} to {default}."
+                )
+                transformed = transform_color(default)
+        return transformed
+
+    def _tile_colors(self, colors: ColorType) -> np.ndarray:
+        """Takes an input color array and forces into being the length of self.data.
+        Used when a single color is supplied for many input points, but we need
+        self.face_color \\ self.edge_color to have the shape of the actual data.
+
+        Parameters
+        --------
+        color : ColorType
+            The user's input after being normalized by self._transform_color
+
+        Returns
+        -----
+        tiled : np.ndarray
+            A tiled version (if needed) of the original input
+        """
+        data_len = len(self.data)
+        # len == 0 data is handled somewhere else
+        if (len(colors) == data_len) or (data_len == 0):
+            return np.asarray(colors)
```

Comment:
This function should never receive raw user input, it will always receive the output of `transform_color_with_defaults` (as the docstring says, which I now made more explicit). It shouldn't be able to handle list of strings, only proper Nx4 arrays. The only functions that know what to do with colors which aren't arrays are `transform_color_with_defaults` and `transform_color`.